lib/protocol/protocol.go

```diff
@@ -51,14 +52,20 @@ const (
 	SymlinkTypeMask = FlagDirectory | FlagSymlinkMissingTarget
 )
 
-// IndexMessage message flags (for IndexUpdate)
+// Request message flags
 const (
-	FlagIndexTemporary uint32 = 1 << iota
+	FlagFromTemporary uint32 = 1 << iota
 )
 
-// Request message flags
+// FileDownloadProgressUpdate update types
+const (
+	UpdateTypeAppend uint32 = iota
+	UpdateTypeForget
+)
+
+// CLusterCOnfig flags
 const (
-	FlagRequestTemporary uint32 = 1 << iota
+	FlagClusterConfigTemporaryIndexesDisabled uint32 = 1 << 0
```

Comment:
Why not make this just `FlagClusterConfigTemporaryIndexes`, with it being set meaning it is supported? That gives some conceptual compatibility with the current messages sent as they send this bit as a zero and don't support temporary indexes.